common/src/main/java/io/pravega/common/util/ByteArraySegment.java

```diff
@@ -169,6 +171,45 @@ public void copyTo(OutputStream stream) throws IOException {
         stream.write(this.array, this.startOffset, this.length);
     }
 
+    @Override
+    @SneakyThrows(IOException.class)
+    public boolean equals(BufferView other) {
+        if (this.length != other.getLength()) {
+            return false;
+        } else if (other instanceof ArrayView) {
+            return equals((ArrayView) other);
+        }
+
+        InputStream otherReader = other.getReader();
+        for (int i = 0; i < this.length; i++) {
+            if ((byte) otherReader.read() != this.array[this.startOffset + i]) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean equals(ArrayView other) {
+        if (this.length != other.getLength()) {
+            return false;
+        }
+
+        byte[] otherArray = other.array();
+        int otherOffset = other.arrayOffset();
+        for (int i = 0; i < this.length; i++) {
```

Comment:
`Arrays.equals` does this (and takes offsets)

Reply:
There is no overload for `Arrays.equals(byte[], byte[])` that takes offsets. At least not in Java 8 which is what we are targeting.

Reply:
See #4884. We'll tackle that once we target Java 11.